unfoldingWord/translationCore
# Finished Localization

#### This pull request addresses:

This finishes #2265.
The updates in this PR mostly involve reducers and actions. A number of these were refactored to eliminate dead code and add support for localization.

* All localization has been moved outside of the reducers.
* Where convenient, or where necessary for the reducers' sake, localization was moved outside of actions.
* Where inconvenient, localization has been left inside actions. It is my opinion that localization should be restricted to the components/containers however, for the sake of time and not rewriting to many things I exposed a method for using locale within actions (thunks). I did label it as `@deprecated` and it's use should be discouraged. I would suggest slowly migrating away from it's use.

> Note:
> * Names for books of the Bible have **not** been configured for localization.
> * Support for localizing tools has **not** been added yet.

#### How to test this pull request:

Make sure everything still works!

Of particular note please double check:

* The home and project steppers. To say these were changed would be an understatement. They were radically overhauled to support localization.
* As part of the above point, the home screen navigation and stepper components were heavily refactored.




---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/unfoldingword-dev/translationcore/3742)
